src/System.Private.Xml.Linq/src/System/Xml/Linq/XNamespace.cs

```diff
@@ -17,7 +17,6 @@ public sealed class XNamespace
         internal const string xmlnsPrefixNamespace = "http://www.w3.org/2000/xmlns/";
 
         private static XHashtable<WeakReference> s_namespaces;
-        private static WeakReference s_refNone;
         private static WeakReference s_refXml;
         private static WeakReference s_refXmlns;
```

Comment:
@krwq as we talked on teams not only `s_refNone` but all these 3 static namespaces could cause the issue, I am not sure about why they make these namespaces static but all others keeping in `s_namespaces;` table as WeakReference. I guess because of performance, so I did performance test and this change is degrading perf some level. 

I could have another solution which is make below hashtable `private readonly XHashtable<XName> _names;` to be WeakReference. I have implemented that solution https://github.com/buyaa-n/corefx/pull/5 and tested performance. It still degrading performance but better than this solution so I gonna go with this https://github.com/buyaa-n/corefx/pull/5, I just need to add some test cases to this solution

Reply:
@buyaa-n how much does it degrade perf?

Reply:
I have got little different results each time, seems the result depend on how much resource am using, need to run perf in very similar/stable env
I thinks this is my most stable result with this solution:
worse: 7, geomean: 1.531
total diff: 7

| Slower                                                  | diff/base | Base Median (ns) | Diff Median (ns) | Modality|
| ------------------------------------------------------- | ---------:| ----------------:| ----------------:| --------:|
| System.Xml.Linq.Perf_XName.GetLocalName                 |      1.88 |            26.58 |            50.02 |         |
| System.Xml.Linq.Perf_XElement.GetAttribute              |      1.84 |            27.86 |            51.27 |         |
| System.Xml.Linq.Perf_XName.GetLocalNameFromExpandedName |      1.71 |            31.75 |            54.23 |         |
| System.Xml.Linq.Perf_XElement.GetElement                |      1.57 |            38.98 |            61.22 |         |
| System.Xml.Linq.Perf_XElement.CreateElement             |      1.51 |            46.76 |            70.77 |         |
| System.Xml.Linq.Perf_XElement.CreateWithElements        |      1.30 |            88.15 |           114.26 |         |
| System.Xml.Linq.Perf_XDocument.Parse                    |      1.08 |          2630.12 |          2840.10 |         |

Reply:
This is the result running perf with other solution:
worse: 6, geomean: 1.109
total diff: 6

| Slower                                               | diff/base | Base Median (ns) | Diff Median (ns) | Modality|
| ---------------------------------------------------- | ---------:| ----------------:| ----------------:| --------:|
| System.Xml.Linq.Perf_XElement.GetAttribute           |      1.17 |            27.86 |            32.71 |         |
| System.Xml.Linq.Perf_XName.GetLocalName              |      1.16 |            26.58 |            30.83 |         |
| System.Xml.Linq.Perf_XElement.GetElement             |      1.12 |            38.98 |            43.71 |         |
| System.Xml.Linq.Perf_XElement.CreateElement          |      1.11 |            46.76 |            51.69 |         |
| System.Xml.Linq.Perf_XElement.GetValue               |      1.06 |            51.79 |            55.08 |         |
| System.Xml.Linq.Perf_XName.NonEmptyNameSpaceToString |      1.04 |            29.91 |            30.98 |         |

Reply:
Do you have ideas how much regression there is on a larger XML? Are the numbers as above or smaller/larger?

Reply:
I haven't really tried with large files